cmd/prysmctl/validator/proposer_settings.go

```diff
@@ -0,0 +1,112 @@
+package validator
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/prysmaticlabs/prysm/v4/api/client"
+	"github.com/prysmaticlabs/prysm/v4/api/client/validator"
+	"github.com/prysmaticlabs/prysm/v4/cmd/validator/flags"
+	"github.com/prysmaticlabs/prysm/v4/config/params"
+	validatorType "github.com/prysmaticlabs/prysm/v4/consensus-types/validator"
+	"github.com/prysmaticlabs/prysm/v4/io/file"
+	"github.com/prysmaticlabs/prysm/v4/io/prompt"
+	validatorpb "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1/validator-client"
+	log "github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+	"go.opencensus.io/trace"
+)
+
+func getProposerSettings(c *cli.Context, r io.Reader) error {
+	ctx, span := trace.StartSpan(c.Context, "prysmctl.getProposerSettings")
+	defer span.End()
+	if !c.IsSet(ValidatorHostFlag.Name) {
+		return fmt.Errorf("no --%s flag value was provided", ValidatorHostFlag.Name)
```

Comment:
The same error building already exists in two other places. You can create an `errNoFlag` helper function and reuse it.